website_sale_show_stock_preview/static/src/js/website_sale_show_stock_preview.js

```diff
@@ -0,0 +1,49 @@
+odoo.define('website_sale_show_stock_preview.shop_stock', function (require) {
+'use strict';
+
+var publicWidget = require('web.public.widget');
+var core = require('web.core');
+var _t = core._t;
+var rpc = require('web.rpc');
+var session = require('web.session');
+var ajax = require('web.ajax');
+
+
+publicWidget.registry.WebsiteSaleShowStockPreview = publicWidget.Widget.extend({
+    selector: '#products_grid',
+    template: 'website_sale_show_stock_preview.preview_stock',
+    xmlDependencies: ['/website_sale_show_stock_preview/static/src/xml/website_sale_show_stock_preview.xml'],
+
+    start: function () {
+        var self = this;
+        var def = [this._super.apply(this, arguments)];
+        this.render_stock();
+        return def;
+    },
+    render_stock: function(){
+        var self = this;
+        var products = $(".o_wsale_product_grid_wrapper");
+        var product_ids = [];
+        _.each(products, function(elem) {
+            var image = $(elem).find('img')[0].src;
+            product_ids.push(parseInt(image.split('/')[6]));
+        });
+
+        return this._rpc({
+            route: '/sale/get_combination_info_stock_preview/',
+            params: {'product_template_ids': product_ids},
+        }).then( (products_qty) => {
+            _.each(products_qty, function(product) {
+                const elem = self.get_element_by_id(product.id);
+                elem.parent().parent().parent().parent().find('.product_price')
+                    .append($(core.qweb.render('website_sale_show_stock_preview.preview_stock', {
+                        virtual_available: product.virtual_available,
+                    })).get(0));
+            });
```

Comment:
Following the previous suggestions...
```suggestion
            for (const product of products_qty) {
                $(product_ids[product.id]).find('.product_price')
                    .append($(core.qweb.render('website_sale_show_stock_preview.preview_stock', {
                        virtual_available: product.virtual_available,
                    })).get(0));
            });
```